src/main/RemoteRequest.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * RemoteRequest is used to for http requests on a remote server which can be
+ * fetched in chunks, e.g. using a Range request.
+ * @flow
+ */
+'use strict';
+
+import Q from 'q';
+
+type Chunk = {
+  start: number;
+  stop: number;
+  buffer: Object; // TODO: more efficient data store
```

Comment:
I think I'm missing some context here. How is this distinct from `src/main/RemoteFile.js`?